### PR TITLE
generic velo artifact for finding web shells

### PR DESCRIPTION
**Pull request by sec-hbaer**
Hi,

I created an artifact to detect web shells on windows and linux systems.
It builds on simple data which should always be available:
* look in the web root for web shells with a yara rule (taken from https://github.com/nsacyber/Mitigating-Web-Shells/ )
* filter the web root for file modifications and file creations for files with extensions implying file types usable for web shells
* (windows only) check for sub processes for programs usable in system reconnaissance, started from the web server process

The artifact will probably create many false positives which will need to be manually filtered.
The artifact could in the future be extended with more specific ways of finding web shells.

Best Regards

### CODE REVIEW

File: content/exchange/artifacts/Generic.Detection.WebShells.yaml [hunk 1]
```diff
@@ -0,0 +1,778 @@
...
+      ) 
+      LET webroot_dir <= webroot[0].Dir
+      SELECT * FROM glob(globs=webroot_dir, accessor="auto")
+      WHERE 
+        NOT IsDir 
+        AND if(condition=DateAfter,
+                   then=Btime >= timestamp(string=DateAfter),
+                   else=true)
+        AND Name =~ WebshellRegex
+  - name: FileModification
```
Comment (mgreen27) [inline content/exchange/artifacts/Generic.Detection.WebShells.yaml]: I would suggest joining FileCreation and FileModification scopes to have a single scope showing files of interest between B time and M time. Also add in a DateBefore time too so you can box time effectively.

Check out some of the Windows.NTFS.MFT for good way to do this.

File: content/exchange/artifacts/Generic.Detection.WebShells.yaml [hunk 2]
```diff
@@ -0,0 +1,778 @@
...
+        windows={SELECT WindowsWebRoot AS Dir FROM info() WHERE OS = "windows"},
+        linux={SELECT LinuxWebRoot AS Dir FROM info() WHERE OS = "linux"}
+      ) 
+      LET webroot_dir <= webroot[0].Dir
+      Select * FROM Artifact.Generic.Detection.Yara.Glob(PathGlob=webroot_dir, YaraRule=YaraRule, DateAfter=DateAfter)
+  - name: WindowsProcessCreation
+    precondition:
+      SELECT OS From info() where OS = 'windows'
+    query: |
+      SELECT * FROM Artifact.Windows.EventLogs.Evtx(EvtxGlob='%SystemRoot%\\System32\\winevt\\Logs\\Security\.evtx', IDRegex="4688")
```
Comment (mgreen27) [inline content/exchange/artifacts/Generic.Detection.WebShells.yaml]: Might consider using Velociraptor sigma capability released version 0.7.1 to help with this part.
https://github.com/Velocidex/velociraptor-sigma-rules

Should help you cover evtx (security, sysmon) and using process tracker sources.

File: content/exchange/artifacts/Generic.Detection.WebShells.yaml (hunk 1 above)
Comment (sec-hbaer) [reply content/exchange/artifacts/Generic.Detection.WebShells.yaml]: Ill see about this

File: content/exchange/artifacts/Generic.Detection.WebShells.yaml (hunk 2 above)
Comment (sec-hbaer) [reply content/exchange/artifacts/Generic.Detection.WebShells.yaml]: When I created the artifact, Sigma rules were not supported yet.
Furthermore, I want to support these older Velo versions.

**Comment by mgreen27** [comment]
EDIT - I saw your regex was only for specifying files to look at for time bounds. So thats good.

Could you add in Sysmon to the evtx. Something like:
EvtxGlob: `%SystemRoot%\System32\winevt\Logs\{*Sysmon*,Security}.evtx`
IDRegex=`1|4688`

```
WHERE 
	( Channel =~ 'sysmon' AND EventID = 1 AND EventData.ParentImage =~ WebserverProcessRegex AND EventData.Image =~ SpawnedProcessRegex )
	OR 
	( Channel =~ 'Security' AND EventID = 4688 AND EventData.ParentProcessName =~ WebserverProcessRegex AND EventData.NewProcessName =~ SpawnedProcessRegex )
```

**Pull request merged**